### PR TITLE
Remove get_stubbed_channel

### PR DESCRIPTION
**Pull request by jerith**
`get_stubbed_channel()` is an old test helper that relies on implementation details of `fake_amqp` that are changing soon. It's easily replaced (in the few tests that use it) by a couple of lines that construct an actual channel from a fake client.

### CODE REVIEW

File: vumi/blinkenlights/metrics.py [hunk 1]
```diff
@@ -75,7 +75,8 @@ def stop_polling(self):
         Stop the metric polling and publishing task.
         """
         if self._task:
-            self._task.stop()
+            if self._task.running:
+                self._task.stop()
```
Comment (hodgestar) [inline vumi/blinkenlights/metrics.py]: Seems a bit unrelated to this issue, although seems like a good idea? Just checking it didn't sneak in unintentionally.

**Comment by hodgestar** [comment]
One question, but otherwise :+1:.

File: vumi/blinkenlights/metrics.py (hunk 1 above)
Comment (jerith) [inline vumi/blinkenlights/metrics.py]: One of the tests now tries to stop a task that was never started (because we're adding cleanup that wasn't there before) and that broke the manager teardown.

**Comment by hodgestar** [comment]
:+1:
